### PR TITLE
Remove lines accessor

### PR DESCRIPTION
**Pull request by schneiderderek**
It looked like `SqlFootprint.lines` was only used for testing purposes. I've removed that accessor and reworked the specs to not rely on it.

### CODE REVIEW

File: spec/sql_footprint_spec.rb [hunk 1]
```diff
@@ -14,20 +14,30 @@
...
     end
 
+    let(:footprint_sql) do
+      described_class.stop
+      File.read 'footprint.sql'
+    end
+
+    let(:sql_lines) do
+      footprint_sql.split("\n")
+    end
```
Comment (mikegee) [inline spec/sql_footprint_spec.rb]: Maybe these two should be plain methods, since they shouldn't be called early?

Comment (schneiderderek) [inline spec/sql_footprint_spec.rb]: @mikegee I used `let`s here to make sure the value was memoized.

Comment (mikegee) [inline spec/sql_footprint_spec.rb]: Only the spec I don't like down there calls one twice.

File: spec/sql_footprint_spec.rb [hunk 2]
```diff
@@ -54,21 +64,20 @@
     end
 
     it 'dedupes the same sql' do
-      expect do
-        Widget.create!
-        Widget.create!
-      end.to change { described_class.lines.length }.by(+1)
+      Widget.create!
+      Widget.create!
+      expect(sql_lines).to eq(sql_lines.uniq)
```
Comment (mikegee) [inline spec/sql_footprint_spec.rb]: Can we count the occurrences of `INSERT INTO widgets`? This matches the implementation a bit to much for my liking.

Comment (schneiderderek) [inline spec/sql_footprint_spec.rb]: @mikegee Yeah, I like that better

File: spec/sql_footprint_spec.rb (hunk 1 above)
Comment (schneiderderek) [inline spec/sql_footprint_spec.rb]: @mikegee That has been rewritten now

**Comment by mikegee** [comment]
I like the way you rewrote those specs. They are cleaner now.

But I disagree with the premise that we shouldn't have features in the implementation that are only for the tests. Increasing testability is the same as increasing usability. Maybe someone will want to get at that array of lines from client app?

This change increases the coupling between the part that captures sql statements and the part that writes to a file. I've been thinking about making the file writing part optional. Maybe a client app will want to do something with those sql statements and then persist them itself?

I hope some of that makes sense. Please let me know what you think.

**Comment by schneiderderek** [comment]
@mikegee Thanks.

Originally, I didn't like the lines accessor because it was sorting the sql statements every time it was accessed. Because we had changed the underlying data structure from an `Array` to a `Set` all of the test cases would have needed to convert the `Set` to an `Array` first if I wanted to change the lines accessor to not sort. That didn't seem right, though. So I just ended up dropping it. I do think this is something that we should have in there, but maybe we should design an intermediate container object for handling the sql statements. This would allow us to change the underlying structure without breaking the public interface. The container could also handle stuff line anonymizing the sql as it's added in. 

Does that make sense?

**Comment by mikegee** [comment]
> Does that make sense?

Yup. :heart:

**Comment by schneiderderek** [comment]
This PR is being closed in favor of this one: https://github.com/covermymeds/sql_footprint/pull/15

**Pull request closed without merging**